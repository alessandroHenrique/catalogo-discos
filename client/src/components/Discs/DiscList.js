import React from 'react';
import PropTypes from 'prop-types';
import uuid from 'uuid';
import Disc from './Disc'

class DiscList extends React.Component {
  static propTypes = {
    discs: PropTypes.array
  }

  render() {
    return (
      <ul className="list-group d-flex flex-row flex-wrap justify-content-between">
        {
          this.props.discs.map((disc) => (
            <Disc
              title={disc.title}
              author={disc.author}
              key={uuid()}
              deleteDisc={() => this.props.deleteDisc(disc.id, disc.disc_collection_id)}
              updateDisc={(newInfo) => this.props.updateDisc({...newInfo, id: disc.id, disc_collection_id: disc.disc_collection_id})}
            />
          ))
        }
      </ul>
    )
  }
}

export default DiscList;
