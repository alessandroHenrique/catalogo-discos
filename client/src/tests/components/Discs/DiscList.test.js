import DiscList from '../../../components/Discs/DiscList';
import React from 'react';
import Store from '../../../store/configureStore';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

const store = Store();

describe('DiscList', () => {
  it('should render a DiscList', () => {

    const component = renderer.create(
      <Provider store={store}>
        <DiscList
          discs={[]}
        />
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
