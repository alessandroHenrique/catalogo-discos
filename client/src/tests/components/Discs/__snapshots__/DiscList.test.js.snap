// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscList should render a DiscList 1`] = `
<ul
  className="list-group d-flex flex-row flex-wrap justify-content-between"
/>
`;
